lapce-core/src/language.rs

```diff
@@ -376,6 +388,66 @@ const LANGUAGES: &[SyntaxProperties] = &[
         code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
         extensions: &["hs"],
     },
+    #[cfg(feature = "lang-glimmer")]
+    SyntaxProperties {
+        id: LapceLanguage::Glimmer,
+        language: tree_sitter_glimmer::language,
+        highlight: tree_sitter_glimmer::HIGHLIGHTS_QUERY,
+        comment: "{{!",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["hbs"],
+    },
+    #[cfg(feature = "lang-haxe")]
+    SyntaxProperties {
+        id: LapceLanguage::Haxe,
+        language: tree_sitter_haxe::language,
+        highlight: tree_sitter_haxe::HIGHLIGHTS_QUERY,
+        comment: "//",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["hx"],
+    },
+    #[cfg(feature = "lang-hcl")]
+    SyntaxProperties {
+        id: LapceLanguage::HCL,
+        language: tree_sitter_hcl::language,
+        highlight: tree_sitter_hcl::HIGHLIGHTS_QUERY,
+        comment: "//",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["hcl"],
+    },
+    #[cfg(feature = "lang-ocaml")]
+    SyntaxProperties {
+        id: LapceLanguage::OCaml,
+        language: tree_sitter_ocaml::language_ocaml,
+        highlight: tree_sitter_ocaml::HIGHLIGHTS_QUERY,
+        comment: "(*",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["ml"],
+    },
+    #[cfg(feature = "lang-ocaml")]
+    SyntaxProperties {
+        id: LapceLanguage::OCaml,
+        language: tree_sitter_ocaml::language_ocaml_interface,
+        highlight: tree_sitter_ocaml::HIGHLIGHTS_QUERY,
+        comment: "(*",
+        indent: "  ",
+        code_lens: (DEFAULT_CODE_LENS_LIST, DEFAULT_CODE_LENS_IGNORE_LIST),
+        extensions: &["mli"],
+    },
+    #[cfg(feature = "lang-scss")]
+    SyntaxProperties {
+        id: LapceLanguage::SCSS,
+        language: tree_sitter_scss::language,
+        highlight: tree_sitter_scss::HIGHLIGHTS_QUERY,
+        comment: "//",
```

Comment:
SCSS also has support for comments like `/* text here */` but our current system doesn't allow for multiple comment types.

Reply:
> Comments really need to be reworked for Lapce. We could just let that field take a closure that returns string slices per a given string.

Now that I think about, this comment field is probably used for automatically commenting things with a shortcut, isn't it? In that case, one comment only makes sense but still isn't preferable honestly, we might wanna have a thing where it uses one comment type for multi line comments and vice versa. But all of this still is a problem with languages that have comments with two symbols on the sides of your text.